src/medCore/gui/viewContainers/medViewContainer.cpp

```diff
@@ -829,3 +859,129 @@ void medViewContainer::removeColorIndicator(QColor color)
 {
     d->poolIndicator->removeColorIndicator(color);
 }
+
+/**
+ * @brief saves the scene in a XML file
+ * Saves views (all layers), and toolboxes parameters
+ * 	expected tree is as follow:
+ *  workingDir (user-defined)
+ * 		|-viewID0
+ * 			|-mapping.xml
+ * 			|-layerID01.xml
+ * 			|-layerID02.xml
+ * 			|-data_file (as saved by the dedicated writer)
+ */
+QString medViewContainer::saveScene()
+{
+    QString dirPath = QFileDialog::getExistingDirectory(this, tr("Open Directory"),"/home",QFileDialog::ShowDirsOnly);
+    QDir workingDir(dirPath);
+    QDomDocument doc("xml");
+    QDomElement root=doc.createElement("scene");
+    doc.appendChild(root);
+
+    QString subDirName;
+
+    medAbstractLayeredView* layeredView=dynamic_cast<medAbstractLayeredView*>(view());
+    if (layeredView==0)
+    {
+        qDebug()<< Q_FUNC_INFO <<", dynamic_cast<medAbstractLayeredView*>(view()) failed";
+        medMessageController::instance()->showError("Scene saving failed", 5000);
+        return "";
+    }
+    else
+    {
+        subDirName="view"+QUuid::createUuid ().toString();
+        if(!workingDir.mkdir(subDirName))
+        {
+            QString errorMessage = ", Failed to create new directory: " + subDirName;
+            qDebug()<< Q_FUNC_INFO  << errorMessage;
+            medMessageController::instance()->showError(errorMessage, 5000);
+        }
+        workingDir.cd(subDirName);
+        QString generatedPath=workingDir.canonicalPath()+"/mapping.xml";
+        QDomElement layeredViewInfo=doc.createElement("layeredView");
+        layeredViewInfo.setAttribute("path",subDirName+"/mapping.xml");
+        layeredViewInfo.setAttribute("id",layeredView->identifier());
+        root.appendChild(layeredViewInfo);
+        layeredView->write(generatedPath);
+        workingDir.cdUp();
+    }
+    QString generatedPath=workingDir.canonicalPath()+"/globalMapping.xml";
+
+    QFile file(generatedPath);
+    if (!file.open(QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate))
+    {
+        return "";
+    }
+    else
+    {
+        QTextStream out(&file);
+        out << doc.toString();
+        QString path = dirPath + "/" + subDirName;
+        return path;
+    }
```

Comment:
This is one of the places were I actually prefer exceptions. We can handle it like this, but it's a lot of code to write. I am not, however, advocating the introduction of exceptions - medInria wasn't designed with them in mind, so it'd be a bad idea to introduce them now.

I think this approach, overall, is probably the best one - there are a number of things that can go wrong when dealing with the system outside medInria, and many of them can be the fault of the user. More specific error messages are therefore better here - so I agree with mm.